virtual_machine.go

```diff
@@ -495,21 +497,23 @@ func (v *VirtualMachine) AgentExecStatus(pid int) (status *AgentExecStatus, err
 }
 
 func (v *VirtualMachine) WaitForAgentExecExit(pid, seconds int) (*AgentExecStatus, error) {
-	wait := time.Duration(seconds) * time.Second
+	timeout := time.After(time.Duration(seconds) * time.Second)
+	ticker := time.NewTicker(100 * time.Millisecond)
```

Comment:
I just reduced the ticker interval here to 100ms instead of the default 1sec, which feels a bit long for this sort of thing

Reply:
that's fair... maybe make a new exported var for DefaultAgentWaitInterval and make it 100ms? then a consumer can modify it instead of having it hard coded in the func

Reply:
I've done that now.  Just as complete aside, I think it's slightly nicer to use the optional function parameter pattern ([example blog post](https://levelup.gitconnected.com/optional-function-parameter-pattern-in-golang-c1acc829307b)) for that sort of thing – slightly more conventional/idiomatic, but also a bit nicer not to be messing with global variables.  However, it would feel out of place to stick that in here without adopting the same elsewhere in the API.

Reply:
the with option thing is done when you create the client. I don't think most people will change the default interval as the values are perfectly sane and normal but you can if you need too. I agree that each func could need a different interval and someday this is where it goes but today I would take the exported var over hardcoded in the func